lerobot/common/robot_devices/cameras/opencv.py

```diff
@@ -219,6 +219,7 @@ def __init__(self, camera_index: int, config: OpenCVCameraConfig | None = None,
         self.camera = None
         self.is_connected = False
         self.thread = None
+        self.lock = threading.RLock()
```

Comment:
```suggestion
        # Using Lock to avoid race condition and segfault when multiple threads
        # access the same camera. This is not one of our use case, but we add this
        # for safety if users want to use this class with threads. As a result, threads
        # will go sequentially through the code logic protected by a lock, instead of 
        # in parallel. Also, we use Recursive Lock to avoid deadlock by allowing each
        # thread to acquire the lock multiple times.
        # TODO(rcadene, aliberts): Add RLock on every robot devices where it makes sense?
        self.lock = threading.RLock()
```